llm-examples/mpt/mpt-7b/MLFlow Logging and inference with MPT-7B model.py

```diff
@@ -0,0 +1,212 @@
+model_location = snapshot_download(repo_id="mosaicml/mpt-7b-instruct", cache_dir="/local_disk0/.cache/huggingface/", revision="bbe7a55d70215e16c00c1825805b81e4badb57d7")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC Log the model to MLFlow
+
+# COMMAND ----------
+
+from mlflow.models.signature import ModelSignature
+from mlflow.types import DataType, Schema, ColSpec
+
+# Define input and output schema
+input_schema = Schema([
+    ColSpec(DataType.string, "prompt"), 
+    ColSpec(DataType.double, "temperature"), 
+    ColSpec(DataType.long, "max_tokens")])
+output_schema = Schema([ColSpec(DataType.string)])
+signature = ModelSignature(inputs=input_schema, outputs=output_schema)
+
+# Define input example
+input_example=pd.DataFrame({
+            "prompt":["what is ML?"], 
+            "temperature": [0.5],
+            "max_tokens": [100]})
+
+# Log the model with its details such as artifacts, pip requirements and input example
+# This may take about 5 minutes to complete
+with mlflow.start_run() as run:  
+    mlflow.pyfunc.log_model(
+        "model",
+        python_model=MPT(),
+        artifacts={'repository' : model_location},
+        pip_requirements=[f"torch=={torch.__version__}", 
+                          f"transformers=={transformers.__version__}", 
+                          f"accelerate=={accelerate.__version__}", "einops", "sentencepiece"],
+        input_example=input_example,
+        signature=signature
+    )
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC Load the model from MLFlow and run batch inference.
+
+# COMMAND ----------
+
+import mlflow
+import pandas as pd
+logged_model = "runs:/"+run.info.run_id+"/model"
+
+# Load model as a PyFuncModel.
+loaded_model = mlflow.pyfunc.load_model(logged_model)
+
+# Predict on a Pandas DataFrame.
+
+input_example=pd.DataFrame({"prompt":["what is ML?", "Name 10 colors."], "temperature": [0.5, 0.5],"max_tokens": [100, 100]})
+print(loaded_model.predict(input_example))
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Register the model
+
+# COMMAND ----------
+
+# Register model in MLflow Model Registry
+# This may take 7 minutes to complete
+result = mlflow.register_model(
+    "runs:/"+run.info.run_id+"/model",
+    name="mpt-7b-instruct",
+    await_registration_for=1000,
+)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Load the model from model registry
+# MAGIC Assume that the below code is run separately or after the memory cache is cleared.
+# MAGIC You need to cleanup the GPU memory.
+
+# COMMAND ----------
+
+!pip install pynvml
+
+# COMMAND ----------
+
+from pynvml import * 
+def print_gpu_utilization(): 
+  nvmlInit() 
+  handle = nvmlDeviceGetHandleByIndex(0) 
+  info = nvmlDeviceGetMemoryInfo(handle)
+  print(f"GPU memory occupied: {info.used//1024**2} MB.") 
+
+print_gpu_utilization() 
+from numba import cuda 
+device = cuda.get_current_device() 
+device.reset() 
```

Comment:
This might cause CUDA error: go/dogfood/#notebook/1647208612764761/command/1647208612764782